tools/probes/probes_main.c

```diff
@@ -182,39 +183,44 @@ void finalize_wave_files(struct wave_files *files)
 	}
 }
 
-int validate_data_packet(struct probe_data_packet *data_packet)
+int validate_data_packet(struct probe_data_packet *packet)
 {
-	uint32_t received_crc;
-	uint32_t calc_crc;
-
-	received_crc = data_packet->checksum;
-	data_packet->checksum = 0;
-	calc_crc = crc32(0, (char *)data_packet, sizeof(*data_packet));
-
-	if (received_crc == calc_crc) {
-		return 0;
-	} else {
-		fprintf(stderr, "error: data packet for buffer %d is not valid: crc32: %d/%d\n",
-			data_packet->buffer_id, calc_crc, received_crc);
+	uint64_t *checksump;
+	uint64_t sum;
+
+	sum = (uint32_t) (packet->sync_word +
+			  packet->buffer_id  +
+			  packet->format +
+			  packet->timestamp_high +
+			  packet->timestamp_low +
+			  packet->data_size_bytes);
+
+	checksump = (uint64_t *) (packet->data + packet->data_size_bytes);
+
+	if (sum != *checksump) {
+		fprintf(stderr, "Checksum error 0x%016lx != 0x%016lx\n", sum, *checksump);
```

Comment:
is `lx` enough to print 64 bits? Don't you need `llx`?

Reply:
Apparently it is. I got a warning when I first tried llx. I think both long long int and long int are 64-bit in 64-bit x86 architecture, and it depends on how the int64_t is defined whether llx or lx should de used. Pretty annoying actually.

Reply:
https://en.cppreference.com/w/cpp/header/cinttypes There's the <inttypes.h> header that provides some nice printf format string macros that work well when you're not trying to do any padding or other custom formatting. See if they are useful. (for example, PRIx64).

Reply:
The macro appear to be work with padding too. Thanks, will update.